source/MaterialXContrib/MaterialXNode/CreateMaterialXNodeCmd.cpp

```diff
@@ -15,25 +15,93 @@
 #include <maya/MFnTypedAttribute.h>
 #include <maya/MGlobal.h>
 
+#include <maya/MViewport2Renderer.h>
+#include <maya/MFragmentManager.h>
+
 #include <algorithm>
 #include <sstream>
 
+namespace mx = MaterialX;
+
 namespace
 {
-    const char* const kDocumentFlag     = "d";
-    const char* const kDocumentFlagLong = "document";
+const char* const kDocumentFlag     = "d";
+const char* const kDocumentFlagLong = "document";
+
+const char* const kElementFlag      = "e";
+const char* const kElementFlagLong  = "element";
+
+const char* const kTextureFlag      = "t";
+const char* const kTextureFlagLong  = "texture";
+
+const char* const kOgsXmlFlag       = "x";
+const char* const kOgsXmlFlagLong   = "ogsxml";
+
+void registerFragment(const MaterialXData& materialData, const std::string& ogsXmlPath)
```

Comment:
MaterialXData is not supposed to know about Maya so have moved this out to here, knowing we'll need to do some more reshuffling when we switch to allowing the document/element string from the node.